pkg/hooks/connection/factory.go

```diff
@@ -132,7 +133,9 @@ func capture(db platform.TestCaseDB, req *http.Request, resp *http.Response, log
 			Body:       string(respBody),
 		},
 		// Mocks: mocks,
-	})
+	}, ctx)
+	testTotal := ctx.Value("testsTotal").(*int)
```

Comment:
why are you doing this here can't you do it inside writeTestCase Function..?

Reply:
changed